src/daos/sql-query-builder.php

```diff
@@ -153,13 +167,18 @@ public function get_sql() {
   /**
    * Calls get_sql() to generate the sql and then creates the prepared statement
    * and binds its values.
+   *
    * @param $db : a pdo database connection
    * @return pdo::Statement object with the query and values bound
    */
   public function get_statement( $db ) {
     $statement = $db->prepare( $this->get_sql() );
     foreach ( $this->values_to_bind as $name => $value ) {
-      $statement->bindValue( $name, $value );
+      if ( is_array( $value ) && 2 === count( $value ) ) {
+        $statement->bindValue( $name, $value[0], $value[1] );
+      } else {
+        $statement->bindValue( $name, $value );
+      };
```

Comment:
Could you explain this logic? your depending on the variable type being passed into at `bind_value` as an array. Thats not the clearest implementation.